src/vehicle.cpp

```diff
@@ -5613,7 +5554,7 @@ void vehicle::update_time( const calendar &update_to )
 
     // Weather stuff, only for z-levels >= 0
     // TODO: Have it wash cars from blood?
-    if( funnels.empty() && solar_panels.empty() ) {
+    if( funnels.empty() ) {
```

Comment:
You mean with the "last updated" check or without it? Inlined or not?

Because inlining it all in `idle` sounds very risky, since most of current `idle` is not adapted well to long term processing.